File: src/main/java/com/github/kelin/archetype/entity/CustomerRecord.java
package com.github.kelin.archetype.entity;

import static com.github.kelin.archetype.entity.EntityCollections.CUSTOMER_RECORDS;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = CUSTOMER_RECORDS)
public class CustomerRecord {
    @Id
    public String id;

    public String firstName;

    public String record;

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public CustomerRecord(String firstName, String record) {
        this.firstName = firstName;
        this.record = record;
    }

    public CustomerRecord() {
    }

    public CustomerRecord(String record) {
        this.record = record;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getRecord() {
        return record;
    }

    public void setRecord(String record) {
        this.record = record;
    }
}

File: src/main/java/com/github/kelin/archetype/entity/CustomerRecord.kt
package com.github.kelin.archetype.entity

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document(collection = EntityCollections.CUSTOMER_RECORDS)
class CustomerRecord {
    @Id
    var id: String? = null
    var firstName: String? = null
    @JvmField
    var record: String? = null

    constructor(firstName: String?, record: String?) {
        this.firstName = firstName
        this.record = record
    }

    constructor()
    constructor(record: String?) {
        this.record = record
    }
}
